widgetssdk/src/main/java/com/glia/widgets/survey/SurveyActivity.java

```diff
@@ -0,0 +1,42 @@
+package com.glia.widgets.survey;
+
+import android.os.Bundle;
+
+import androidx.appcompat.app.AppCompatActivity;
+
+import com.glia.androidsdk.engagement.Survey;
+import com.glia.widgets.GliaWidgets;
+import com.glia.widgets.R;
+import com.glia.widgets.di.Dependencies;
+
+public class SurveyActivity extends AppCompatActivity {
+    private static final String TAG = SurveyActivity.class.getSimpleName();
+
+    private SurveyView surveyView;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+        setContentView(R.layout.survey_activity);
+
+        surveyView = findViewById(R.id.survey_view);
+
+        SurveyContract.Controller surveyController =
+                Dependencies.getControllerFactory().getSurveyController();
+        surveyView.setController(surveyController);
+
+        Bundle extras = getIntent().getExtras();
+        Survey survey = extras.getParcelable(GliaWidgets.SURVEY);
```

Comment:
Survey can be non-parcelable

You could pass "survey id" (or identification for pending survey if even needed) with intent to survey activity
and query pending survey on controller init.

So this way Controller would get it from Repository, not the Activity

Reply:
Suggestion from https://github.com/salemove/android-sdk-widgets/pull/259#discussion_r855957103

Reply:
A survey can be got from the engagement object. Is it possible to get already ended engagement?

Reply:
Also, the engagement can not contain a survey. What in this case can we do? Open the survey activity and then finish it?

Reply:
`GliaSurveyRepository.class` repository receives the pending survey object.
You can hold it there instead of cascading down to Chat or Call activity.

example:
SurveyRepository - ResultStorage object is storing the pending survey already
ChatController - `viewCallback.navigateToSurvey(survey);` at line 1180 - remove the "survey" part from callback
`viewCallback.navigateToSurvey();`  <- after that You will see what needs to be removed.

SurveyActivity - remove parcelable survey part
SurveyController remove survey part from init
SurveyController on init could ask SurveyRepository for pending survey object

If survey was null then according to null checks in Chat and Call controller it shouldn't even get to SurveyActivity and would show endengagement dialog instead.

This way You will not need to open/close it :)